File: Tokopedia/Classses/UIRefactor/Home/MyWishlist/MyWishlistShop.m
//
//  MyWhishlistMojitoShop.m
//  Tokopedia
//
//  Created by Billion Goenawan on 9/23/16.
//  Copyright © 2016 TOKOPEDIA. All rights reserved.
//

#import "MyWishlistShop.h"

@implementation MyWishlistShop

    +(NSDictionary *)attributeMappingDictionary
    {
        NSArray *keys = @[@"id",@"name", @"url", @"reputation", @"gold_merchant", @"lucky_merchant", @"location", @"status", @"available", @"status", @"preorder"];
        ;
        return [NSDictionary dictionaryWithObjects:keys forKeys:keys];
    }

    +(RKObjectMapping*)mapping
    {
        RKObjectMapping *mapping = [RKObjectMapping mappingForClass:self];
        [mapping addAttributeMappingsFromDictionary: [self attributeMappingDictionary]];
        return mapping;
    }

@end

File: Tokopedia/Classses/UIRefactor/Home/MyWishlist/MyWishlistShop.swift
//
//  MyWishlistShop.swift
//  Tokopedia
//
//  Created by Billion Goenawan on 10/14/16.
//  Copyright © 2016 TOKOPEDIA. All rights reserved.
//

import UIKit

@objc(MyWishlistShop)
class MyWishlistShop: NSObject {
    
    var id: String!
    var name: String!
    var url: String!
    var reputation: MyWishlistShopReputation!
    var gold_merchant: Bool = false
    var lucky_merchant: String!
    var location: String!
    var status: String!
    
    class func mapping() -> RKObjectMapping {
        let mapping = RKObjectMapping(for: MyWishlistShop.self)
        
        mapping?.addAttributeMappings(from:["id", "name", "url", "gold_merchant", "lucky_merchant", "location", "status"])
        
        mapping?.addPropertyMapping(RKRelationshipMapping(fromKeyPath: "reputation", toKeyPath: "reputation", with: MyWishlistShopReputation.mapping()))
        
        return mapping!
    }
}
